### PR TITLE
👽️(API/Feature) : Login and Registration API Integration

### LINKED ISSUE #200: Improve Hash speed
Tweak parameters to achieve a faster hash and authentication speed

Goal : sub 1 (acceptable for demo 3 sub 3)

### LINKED ISSUE #205: Add login and registration functionality to static pages
Integrate the API into the login and registration pages

### LINKED ISSUE #206: Implement dispatches on login
Store user id and type in the state manger on login 
use the type to navigate to the appropriate dashbaord

### PR DESCRIPTION
**Pull request by priyo5**
Closes #206 
Closes #205 
Closes #200

### CODE REVIEW

File: libs/login/feature/src/lib/login.component.html [hunk 1]
```diff
@@ -3,8 +3,10 @@ <h2>Login</h2>
     <div class="loginDetails">
         <form [formGroup]="loginDetailsForm" (ngSubmit)="loginUser()" class="form-details">
             <the-au-pair-input-field labelContent="Email" [showError]="errState && (!loginDetailsForm.controls['email'].valid)" formControlName="email"></the-au-pair-input-field>
-            
+            <p id="emailError" class="error" style="display: none;">Error in field</p>
```
Comment (Ruben-Brits) [inline libs/login/feature/src/lib/login.component.html:6]: is it possible to move these display none's into the css and not inline

File: libs/register/feature/src/lib/register.component.html [hunk 2]
```diff
@@ -8,24 +8,36 @@ <h2>Register</h2>
                 <input class = "parent" [class]="(this.parentChosen===true) ? 'chosen' : 'notchosen'" type="button" (click)="this.parentChosen = true" value="I am a Parent">
                 <input class = "aupair" [class]="(this.parentChosen===true) ? 'notchosen' : 'chosen'" type="button" (click)="this.parentChosen = false" type="button" value="I am an Au Pair">
             </div>
-                
-            <the-au-pair-input-field labelContent="Name" [showError]="submitAttempt && (!parentRegisterDetailsForm.controls['name'].valid)" formControlName="name"></the-au-pair-input-field>
             
+            <the-au-pair-input-field labelContent="Name" [showError]="submitAttempt && (!parentRegisterDetailsForm.controls['name'].valid)" formControlName="name"></the-au-pair-input-field>
+            <p id="nameError" class="error" style="display: none;">Error in field</p>
```
Comment (Ruben-Brits) [inline libs/register/feature/src/lib/register.component.html:13]: Same here with inline css, please move out to the scss file on the id or class

File: libs/register/feature/src/lib/register.component.spec.ts [hunk 3]
```diff
@@ -11,17 +11,17 @@ describe('RegisterComponent', () => {
...
-  const invalidId = { name: "Test", surname: "van Tester", email: "test@test.com", phone: "1234567890", id: "99999911234098", medAid: "1234", address: "123 Test Road", pass: "TesterPass321!", confPass: "TesterPass321!" };
-  const invalidMedAid = { name: "Test", surname: "van Tester", email: "test@test.com", phone: "1234567890", id: "0101011234098", medAid: "1234!", address: "123 Test Road", pass: "TesterPass321!", confPass: "TesterPass321!" };
-  const invalidPass = { name: "Test", surname: "van Tester", email: "test@test.com", phone: "1234567890", id: "0101011234098", medAid: "1234", address: "123 Test Road", pass: "TesterPass", confPass: "TesterPass321!" };
-  const invalidConfPass = { name: "Test", surname: "van Tester", email: "test@test.com", phone: "1234567890", id: "0101011234098", medAid: "1234", address: "123 Test Road", pass: "TesterPass321!", confPass: "TesterPass321!@" };
+  // //Inalid form
+  // const invalidName = { name: "Test1", surname: "van Tester", email: "test@test.com", phone: "1234567890", id: "0101011234098", medAid: "1234", address: "123 Test Road", pass: "TesterPass321!", confPass: "TesterPass321!" };
```
Comment (Ruben-Brits) [inline libs/register/feature/src/lib/register.component.spec.ts:17]: Can all these comments be removed or are we keeping them for a specif reason?

File: libs/register/feature/src/lib/register.component.ts [hunk 4]
```diff
@@ -10,34 +14,356 @@ export class RegisterComponent {
...
+      }
+    }
+
+    dom = document.getElementById("pswError");
+    if(!this.parentRegisterDetailsForm.controls['pass'].valid)
+    {
+      formError = true
+      if(dom != null)
+      {
+        dom.innerHTML = "Invalid password : should contain upper and lowercase characters aswell as special characters and numbers";
```
Comment (Ruben-Brits) [inline libs/register/feature/src/lib/register.component.ts:220]: aswell -> as well

Also please check consistency with other errors like ending some errors with full stops and others not?

File: libs/register/feature/src/lib/register.component.ts [hunk 5]
```diff
@@ -10,34 +14,356 @@ export class RegisterComponent {
...
+          return;
+        }
 
-    return 0;
+        const len = res.length;
+        for (let j = 0; j < len && j<4; j++) 
+        { 
+          this.potentialLocations.push(res[j].display_name);
+        }
+        console.log(this.potentialLocations);
```
Comment (Ruben-Brits) [inline libs/register/feature/src/lib/register.component.ts:362]: Remove unneeded console logs unless still important for testing purposes please

**Review by Ruben-Brits — CHANGES_REQUESTED** [review]

File: libs/register/feature/src/lib/register.component.ts (hunk 5 above)
Comment (priyo5) [reply libs/register/feature/src/lib/register.component.ts:362]: Used for address fucntionality untill custom selector is made

File: libs/register/feature/src/lib/register.component.spec.ts (hunk 3 above)
Comment (Ruben-Brits) [reply libs/register/feature/src/lib/register.component.spec.ts:17]: Cool, just make  note to remove in future please

File: libs/register/feature/src/lib/register.component.ts (hunk 5 above)
Comment (Ruben-Brits) [reply libs/register/feature/src/lib/register.component.ts:362]: Cool, just make note to remove then when done

**Review by Ruben-Brits — DISMISSED** [review]

**Review by PxttytheThxtty — APPROVED** [review]

**Review by Ruben-Brits — APPROVED** [review]

**Pull request merged**